python/hsfs/storage_connector.py

```diff
@@ -646,3 +635,105 @@ def snowflake_connector_options(self):
         if self._warehouse is not None:
             props["warehouse"] = self._warehouse
         return props
+
+    def spark_options(self):
+        """Return prepared options to be passed to Spark, based on the additional
+        arguments.
+        """
+        props = self._sf_options
+        props["sfURL"] = self._url
+        props["sfSchema"] = self._schema
+        props["sfDatabase"] = self._database
+        props["sfUser"] = self._user
+        if self._password is not None:
+            props["sfPassword"] = self._password
+        else:
+            props["sfAuthenticator"] = "oauth"
+            props["sfToken"] = self._token
+        if self._warehouse is not None:
+            props["sfWarehouse"] = self._warehouse
+        if self._role is not None:
+            props["sfRole"] = self._role
+        if self._table is not None:
+            props["dbtable"] = self._table
+
+        return props
+
+    def read(
+        self,
+        query: str,
+        data_format: str = None,
+        options: dict = {},
+        path: str = None,
+    ):
+        """Reads a query into a dataframe using the storage connector."""
+        options = (
+            {**self.spark_options(), **options}
+            if options is not None
+            else self.spark_options()
+        )
+        if query:
+            options["query"] = query
+
+        return engine.get_instance().read(self, self.SNOWFLAKE_FORMAT, options, None)
+
+
+class JdbcConnector(StorageConnector):
+    type = StorageConnector.JDBC
+    JDBC_FORMAT = "jdbc"
+
+    def __init__(
+        self,
+        id,
+        name,
+        featurestore_id,
+        description=None,
+        # members specific to type of connector
+        connection_string=None,
+        arguments=None,
+    ):
+        super().__init__(id, name, description, featurestore_id)
+
+        # JDBC
+        self._connection_string = connection_string
+        self._arguments = arguments
+
+    @property
+    def connection_string(self):
+        """JDBC connection string."""
+        return self._connection_string
+
+    @property
+    def arguments(self):
+        """Additional JDBC, REDSHIFT, or Snowflake arguments."""
+        return self._arguments
+
+    def spark_options(self):
+        """Return prepared options to be passed to Spark, based on the additional
+        arguments.
+        """
+        args = [arg.split("=") for arg in self._arguments.split(",")]
+
+        options = {a[0]: a[1] for a in args}
+        options["url"] = self._connection_string
+
+        return options
+
+    def read(
+        self,
+        query: str,
+        data_format: str = None,
+        options: dict = {},
+        path: str = None,
+    ):
+        """Reads a query into a dataframe using the storage connector."""
+        # options = (
+        #     {**self.spark_options(), **options}
+        #     if options is not None
+        #     else self.spark_options()
+        # )
```

Comment:
I guess this can be removed.

Reply:
I put it back and removed the single line, not sure why I commented it out